File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/server/filters/filtermethods/TraceFilterSpec.kt
package io.micronaut.docs.server.filters.filtermethods

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe
import io.micronaut.context.ApplicationContext
import io.micronaut.docs.server.intro.HelloControllerSpec
import io.micronaut.http.HttpRequest
import io.micronaut.http.client.HttpClient
import io.micronaut.runtime.server.EmbeddedServer

class TraceFilterSpec: StringSpec() {

    init {
        "test trace filter" {
            val embeddedServer = ApplicationContext.run(EmbeddedServer::class.java,
                mapOf("spec.name" to HelloControllerSpec::class.java.simpleName, "spec.filter" to "TraceFilterMethods", "spec.lang" to "java"))
            val client = embeddedServer.applicationContext.createBean(HttpClient::class.java, embeddedServer.url)

            val response = client.toBlocking().exchange<Any, Any>(HttpRequest.GET("/hello"))

            response.headers.get("X-Trace-Enabled") shouldBe "true"

            embeddedServer.close()
            client.close()
        }
    }
}

File: test-suite-python/src/test/python/micronaut/docs/server/filters/TraceFilterSpec.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.context.annotation import Property
from micronaut.http import HttpRequest
from micronaut.http.client import HttpClient
from micronaut.http.client.annotation import Client
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test


@Property(name="spec.filter", value="TraceFilter")
@MicronautTest
class TraceFilterSpec:
    client: Annotated[HttpClient, Inject, Client("/")]

    @Test
    def testTraceFilter(self):
        response = self.client.toBlocking().exchange(HttpRequest.GET("/hello"))

        assert response.getHeaders().get("X-Trace-Enabled") == "true"
